.github/workflows/bundle.yml

```
@@ -0,0 +1,87 @@
+on: 
+    pull_request:
+      branches: [ main ]
+    push:
+      branches: [ main ]
+name: ci/github
+jobs:
+    checksecret:
+      name: Check if REGISTRY_USERNAME is set in github secrets
+      runs-on: ubuntu-latest
+      outputs:
+        is_registry_username_set: ${{ steps.checksecret_job.outputs.is_registry_username_set }}
+      steps:
+        - name: Check if REGISTRY_USERNAME is set
+          id: checksecret_job
+          env:
+              REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+          run: |
+              echo "is_registry_username_set: ${{ env.REGISTRY_USERNAME != '' }}"
+              echo "::set-output name=is_registry_username_set::${{ env.REGISTRY_USERNAME != '' }}"
+
+    bundle-build:
+        needs: [ checksecret ]
+        if: needs.checksecret.outputs.is_registry_username_set == 'true'
+        strategy:
+          fail-fast: false
+          matrix:
+            go-version: [ 1.16.x ]
+            os: [ ubuntu-latest ]
+            kubernetes:
+              - v1.20.7
+          max-parallel: 2
+        runs-on: ${{ matrix.os }}
+        steps:
+        - name: Install Go
+          uses: actions/setup-go@v2
+          with:
+            go-version: ${{ matrix.go-version }}
+        - name: Check out Code
+          uses: actions/checkout@v2
+        - name: Install kubectl
+          uses: azure/setup-kubectl@v1
+          with:
+            version: ${{ matrix.kubernetes }}
+        - name: Create kind cluster
+          uses: helm/kind-action@v1.1.0
+          with:
+            version: v0.10.0
+            node_image: kindest/node:${{ matrix.kubernetes }}
+            cluster_name: kind
+            wait: 120s
+        - name: Verify kind cluster
+          run: |
+            echo "# Using KinD context..."
+            kubectl config use-context "kind-kind"
+            echo "# KinD nodes:"
+            kubectl get nodes
+        - name: Login to the docker registry
+          env:
+            REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+            REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+          run: echo "$REGISTRY_PASSWORD" | docker login --username "$REGISTRY_USERNAME" --password-stdin quay.io
+        - name: Make and push Operator image
+          env:
+            IMAGE_REPO: quay.io/shipwright
+            TAG: pr-${{ github.event.number }}
+          run: |
+            IMAGE_REPO=ko.local make ko-publish
```

nit - this can be passed as a make variable instead of an env var to the process:
```suggestion
            make ko-publish IMAGE_REPO=ko.local
```